floris/simulation/grid.py

```diff
@@ -79,6 +79,7 @@ class Grid(ABC):
     x_sorted: NDArrayFloat = field(init=False)
     y_sorted: NDArrayFloat = field(init=False)
     z_sorted: NDArrayFloat = field(init=False)
+    cubature_weights: NDArrayFloat = field(init=False, default=None)
```

Comment:
I can see why this is needed here; I wonder whether in future, we should have an abstract TurbineGrid class (inherits from abstract Grid, or possibly general PointsGrid as we have discussed elsewhere) from which TurbineCubatureGrid and TurbineSquareGrid would inherit. Could get a little complicated with multiple levels on inheritance though.

Reply:
I agree. The grid module overall needs some work. It would be helpful to identify where there's overlap between these grid-types. Possibly one path forward is to get rid of one grid super class in favor of categories of grids. At the least, the grid abstract base class could only define the required API's for all grids like `x`, `x_sorted` and `sorted_indeces` etc.